src/main/scala/mesosphere/marathon/api/v2/validation/PodsValidation.scala

```diff
@@ -64,9 +65,84 @@ trait PodsValidation {
     resource.gpus should be >= 0
   }
 
-  val containerValidator: Validator[PodContainer] = validator[PodContainer] { container =>
+  def httpHealthCheckValidator(endpoints: Seq[Endpoint]) = validator[HttpHealthCheck] { hc =>
+    hc.endpoint.length is between(1, 63)
+    hc.endpoint should matchRegexFully(NamePattern)
+    hc.endpoint is isTrue("contained in the container endpoints") { endpoint =>
+      endpoints.exists(_.name == endpoint)
+    }
+    hc.path.map(_.length).getOrElse(1) is between(1, 1024)
+  }
+
+  def tcpHealthCheckValidator(endpoints: Seq[Endpoint]) = validator[TcpHealthCheck] { hc =>
+    hc.endpoint.length is between(1, 63)
+    hc.endpoint should matchRegexFully(NamePattern)
+    hc.endpoint is isTrue("contained in the container endpoints") { endpoint =>
+      endpoints.exists(_.name == endpoint)
+    }
+  }
+
+  val commandCheckValidator = new Validator[CommandHealthCheck] {
+    override def apply(v1: CommandHealthCheck): Result = v1.command match {
+      case ShellCommand(shell) =>
+        (shell.length should be > 0)(shell.length)
+      case ArgvCommand(argv) =>
+        (argv.size should be >0)(argv.size)
+    }
+  }
+
+  def healthCheckValidator(endpoints: Seq[Endpoint]) = validator[HealthCheck] { hc =>
+    hc.gracePeriodSeconds should be >= 0L
+    hc.intervalSeconds should be >= 0
+    hc.maxConsecutiveFailures should be >= 0
+    hc.timeoutSeconds should be >= 0
```

Comment:
Timeout >= 0? Agree on interval. I should have read all of the raml comments on not just the raml restrictions.